varname/ignore.py

```diff
@@ -0,0 +1,343 @@
+                          'as an ignore element for varname.')
+
+    def match(self, frame_no: int, frameinfos: List[inspect.FrameInfo]) -> bool:
+        frame = frameinfos[frame_no].frame
+        return frame.f_code == self.ignore.__code__
+
+    def __repr__(self):
+        return f'<IgnoreFunction({self.ignore.__name__})>'
+
+class IgnoreDecorated(IgnoreElem):
+    """Ignore a decorated function"""
+    def match(self, frame_no: int, frameinfos: List[inspect.FrameInfo]) -> bool:
+        try:
+            frame = frameinfos[frame_no + self.ignore[1]].frame
+        except IndexError:
+            return False
+
+        return frame.f_code == self.ignore[0].__code__
+
+    def __repr__(self):
+        return f'<IgnoreDecorated({self.ignore[0].__name__}, {self.ignore[1]})>'
+
+class IgnoreQualname(IgnoreElem):
+    """Ignore a function by qualified name"""
+    def __init__(self,
+                 ignore: Tuple[Optional[Union[ModuleType, str]], str]) -> None:
+        super().__init__(ignore)
+        self.module_flag = (
+            None if not ignore[0]
+            else 'module' if isinstance(ignore[0], ModuleType)
+            else 'filename'
+        )
+
+        # check uniqueness of qualname
+        self.qname_checked = False
+        if self.module_flag == 'module':
+            modfile = getattr(self.ignore[0], '__file__', None)
+            if modfile and path.isfile(modfile):
+                source = Source.for_filename(modfile, self.ignore[0].__dict__)
+                nobj = list(source._qualnames.values()).count(self.ignore[1])
+                assert nobj == 1, (
+                    f"Qualname {self.ignore[1]!r} in "
+                    f"{self.ignore[0].__name__!r} "
+                    "doesn't exist or refers to multiple objects."
+                )
+                self.qname_checked = True
+
+    def match(self, frame_no: int, frameinfos: List[inspect.FrameInfo]) -> bool:
+        frame = frameinfos[frame_no].frame
+        if self.module_flag == 'module':
+            module = inspect.getmodule(frame)
+            if module and module != self.ignore[0]:
+                return False
+
+            if not module and (
+                    getattr(self.ignore, MODULE_IGNORE_ID_NAME, None) !=
+                    frame.f_globals.get(MODULE_IGNORE_ID_NAME, '')
+            ):
+                return False
+
+            if not self.qname_checked:
+                source = Source.for_frame(frame)
+                if source.tree: # otherwise, no way to check
+                    nobj = list(
+                        source._qualnames.values()
+                    ).count(self.ignore[1])
+                    assert nobj == 1, (
+                        f"Qualname {self.ignore[1]!r} in "
+                        f"{self.ignore[0].__name__!r} "
+                        "doesn't exist or refers to multiple objects."
+                    )
+            return re.match(self.ignore[1],
```

Comment:
Using `re` seem a bit weird. `a.c` will match `abc`. Is this documented? If you want to allow it I think they must pass a compiled pattern.

Reply:
Not yet documented. The `re` is introduced very late, which is used to allow `*.<lambda>`. Probably just say `qualname.endswith('<lambda>')`.

Reply:
In such a case, it is difficult to introduce something like `IgnoreElem(None, '<lambda>')`. Maybe use `fnmatch` instead of `re` to make it simple. Then we can do `IgnoreElem(None, '*<lambda>')`. `re` being too aggressive is one of the reasons that I have documented it yet.